app/src/main/java/app/nameplaceholder/sensorsapp/DataRepository.kt

```diff
@@ -27,6 +27,7 @@ import kotlinx.coroutines.flow.*
 class DataRepository constructor(
     private val context: Context
 ) {
+    private val valuesInMemoryCache = mutableListOf<SensorData>()
```

Comment:
With this refactoring (moving collected data from Flow scope to global one) you start sharing state between multiple invocations. Tracking results should be cleared on tracking `Stop`.